scripts/post_deploy.py

```diff
@@ -29,46 +38,58 @@ def register_with_sam():
     public_tdr.register_with_sam()
 
 
-def verify_sources():
-    tdr_catalogs = {
-        catalog.name
-        for catalog in config.catalogs.values()
-        if catalog.plugins[RepositoryPlugin.type_name()].name == 'tdr'
+def verify_sources(source_names_by_id: Dict[str, str]):
+    dupes = {
+        name
+        for name, count in Counter(source_names_by_id.values()).items()
+        if count > 1
+    }
+    require(not dupes, 'Snapshot names are ambiguous', dupes)
```

Comment:
The goal is to verify that there are no snapshots with different _IDs_ but the same _name_. The immediate reason for doing this is to check that we can sensibly invert the mapping a few lines later. However, the potential pipeline failure you allude to is actually another benefit of this check. If there were snapshots with duplicate names, then the IT is already destined to fail because [`TDRClient._lookup_source` retrieves the snapshot ID by filtering by the snapshot name, and raises `RequirementError` is there is more than one result](https://github.com/DataBiosphere/azul/blob/e1e991e2aadf78f39c4689798a52ca2456a1e084/src/azul/terra.py#L434). So this check helps to catch that earlier and more clearly.

Reply:
That would require an addition import and, more importantly, raise a type warning since `dupes` is not a ` bool`.